packages/mock-block-dock/src/data/entity-types.ts

```diff
@@ -1,61 +1,119 @@
+  type: "object",
+  title: "Company",
+  description: "A company or organization.",
+  properties: {
+    [extractBaseUri(propertyTypes.numberOfEmployees.$id)]: {
+      $ref: propertyTypes.numberOfEmployees.$id,
+    },
+    [extractBaseUri(propertyTypes.name.$id)]: {
+      $ref: propertyTypes.name.$id,
     },
   },
-  {
-    entityTypeId: "Person",
-    schema: {
-      $id: "https://example.com/types/Person",
-      title: "Person",
-
-      type: "object",
-      $schema: "https://json-schema.org/draft/2019-09/schema",
-      description: "A human person.",
-      properties: {
-        age: {
-          type: "number",
-          description: "The age of the person, in years.",
-        },
-        email: {
-          type: "string",
-          description: "An email address.",
-          format: "email",
-        },
-        name: {
-          type: "string",
-          description: "The person's name.",
-        },
-        username: {
-          description: "The person's username in this application",
-          type: "string",
-          minLength: 4,
-          maxLength: 24,
-        },
+  required: [
+    extractBaseUri(propertyTypes.numberOfEmployees.$id),
+    extractBaseUri(propertyTypes.name.$id),
+  ],
+  links: {},
+};
+const person: EntityType = {
+  kind: "entityType",
+  $id: "https://example.com/types/entity-type/person/v/1",
+  type: "object",
+  title: "Person",
+  description: "A human person.",
+  properties: {
+    [extractBaseUri(propertyTypes.age.$id)]: {
+      $ref: propertyTypes.age.$id,
+    },
+    [extractBaseUri(propertyTypes.email.$id)]: {
+      $ref: propertyTypes.email.$id,
+    },
+    [extractBaseUri(propertyTypes.name.$id)]: {
+      $ref: propertyTypes.name.$id,
+    },
+    [extractBaseUri(propertyTypes.username.$id)]: {
+      $ref: propertyTypes.username.$id,
+    },
+  },
+  required: [
+    extractBaseUri(propertyTypes.age.$id),
+    extractBaseUri(propertyTypes.email.$id),
+    extractBaseUri(propertyTypes.name.$id),
+    extractBaseUri(propertyTypes.username.$id),
+  ],
+  links: {
+    [worksFor.$id]: {
+      type: "array",
+      items: {
+        $ref: company.$id,
+      },
+      ordered: false,
+    },
+    [founderOf.$id]: {
+      type: "array",
+      items: {
+        $ref: company.$id,
       },
-      labelProperty: "name",
-      required: ["age", "email", "name", "username"],
+      ordered: false,
     },
   },
-];
+};
+const testType: EntityType = {
+  kind: "entityType",
+  $id: "https://example.com/types/entity-type/test-type/v/1",
+  type: "object",
+  title: "Test Type",
+  description: "A Type for Testing",
+  properties: {
+    [extractBaseUri(propertyTypes.name.$id)]: {
+      $ref: propertyTypes.name.$id,
+    },
+  },
+  required: [extractBaseUri(propertyTypes.name.$id)],
+  links: {},
+};
+
+export const entityTypes = {
+  company,
+  person,
+  worksFor,
+  founderOf,
+  testType,
+};
+// satisfies Record<string, EntityType> };
```

Comment:
I ran into problems with using `satisfies` (can't remember if it was Webpack or something else).

If I could have used it then we could have inlined the `entityTypes` definitions and had them `as const`, which means their type would be preserved. Unfortunately if I type this as `Record<string, EntityType>` then their usages become buggered because TypeScript doesn't agree it's const.

Adding `as const` here also breaks it because the type isn't compatible (`EntityType` isn't `readonly` or something)

Reply:
NextJS [does not yet support](https://github.com/vercel/next.js/discussions/40895) `satisfies`